packages/venia-concept/src/components/Checkout/flow.js

```diff
@@ -1,79 +1,76 @@
+import { bool, func, shape, string } from 'prop-types';
 
 import classify from 'src/classify';
-import Entrance from './entrance';
-import Exit from './exit';
+import Cart from './cart';
 import Form from './form';
+import Receipt from './receipt';
 import defaultClasses from './flow.css';
 
-const stepMap = {
-    READY: 'STEP_1',
-    REQUESTING: 'STEP_1',
-    MODIFYING: 'STEP_2',
-    SUBMITTING: 'STEP_2',
-    ACCEPTED: 'STEP_3'
-};
-
-const stepEnum = Object.keys(stepMap);
-
 class Flow extends Component {
     static propTypes = {
+        actions: shape({
+            editOrder: func.isRequired,
+            resetCheckout: func.isRequired,
+            submitCart: func.isRequired,
+            submitInput: func.isRequired,
+            submitOrder: func.isRequired
+        }).isRequired,
+        checkout: shape({
+            editing: string,
+            step: string,
+            submitting: bool,
+            valid: bool
+        }),
         classes: shape({
             root: string
         }),
-        ready: bool,
-        resetCheckout: func.isRequired,
-        requestOrder: func.isRequired,
-        status: oneOf(stepEnum).isRequired,
-        submitOrder: func.isRequired
+        ready: bool
     };
 
-    render() {
+    get child() {
+        const { actions, checkout, ready } = this.props;
         const {
-            classes,
-            enterSubflow,
-            ready,
+            editOrder,
             resetCheckout,
-            requestOrder,
-            status,
+            submitCart,
+            submitInput,
             submitOrder
-        } = this.props;
-
-        const step = stepMap[status];
-        let child = null;
+        } = actions;
+        const { editing, step, submitting, valid } = checkout;
 
         switch (step) {
-            case 'STEP_1': {
-                child = (
-                    <Entrance status={status} requestOrder={requestOrder} />
-                );
-                break;
+            case 'cart': {
+                const stepProps = { ready, submitCart, submitting };
+
+                return <Cart {...stepProps} />;
             }
-            case 'STEP_2': {
-                child = (
-                    <Form
-                        ready={ready}
-                        status={status}
-                        enterSubflow={enterSubflow}
-                        submitOrder={submitOrder}
-                    />
-                );
-                break;
+            case 'form': {
+                const stepProps = {
+                    editOrder,
+                    editing,
+                    submitInput,
+                    submitOrder,
+                    submitting,
+                    valid
+                };
+
+                return <Form {...stepProps} />;
             }
-            case 'STEP_3': {
-                child = <Exit resetCheckout={resetCheckout} />;
-                break;
+            case 'receipt': {
```

Comment:
This seems like a mismatch. Why's it here in this otherwise very consistent flow, which sets `child` and identifies itself as a step number?

Reply:
Fixed. 👍